pass.go

```diff
@@ -0,0 +1,1054 @@
+
+		// Assume templating is broken, output error string raw to client
+		rw.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintln(rw, "Unexpected error while rendering page!")
+	}
+}
+
+// renderErrorPageJSON does the same as RenderErrorPage, but for json requests
+// only called by RenderErrorPage, do not call directly
+func (p *Pass) renderErrorPageJSON(rw http.ResponseWriter, r *http.Request, errStr string) {
+	rw.Header().Add("Content-Type", "application/json")
+	errorJSON := struct{ Error string }{
+		Error: errStr,
+	}
+
+	if b, err := json.MarshalIndent(errorJSON, "", "  "); err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		log.Printf("renderErrorPageJSON err %v\n", err)
+	} else {
+		rw.Write(b)
+	}
+}
+
+// RenderErrorPage outputs a rendered error template, no output must be made
+// after a call to this function!
+func (p *Pass) RenderErrorPage(rw http.ResponseWriter, r *http.Request, tv interface{}, err error) {
+	var httpStatusCode int
+	var errStr string
+	var innerTemplate *template.Template
+	httpStatusCode, innerTemplate, errStr = p.serverError(err)
+
+	rw.WriteHeader(httpStatusCode)
+
+	if helpers.HTTPAcceptCheck("application/json", r.Header) {
+		p.renderErrorPageJSON(rw, r, errStr)
+		return
+	}
+
+	tv2 := tv.(*TemplateVariables)
+	tv2.ErrorString = errStr
+
+	p.RenderPage(rw, r, innerTemplate, tv2)
+}
+
+// GetURL returns a https link for the configured domain[:port]
+func (p *Pass) GetURL() string {
+	return fmt.Sprintf("https://%s", p.sv.cfg.Hostname)
+}
+
+// GetPassword returns the password stored at the given UUID if any
+func (p *Pass) GetPassword(uuid string) (payload string, err error) {
+	if payload, err = p.sv.db.get(uuid, p.sv.cfg.Secret); err != nil {
+		return payload, err
+	}
+
+	return payload, err
+}
+
+// GetConfigForAddon returns a struct of the unmarshalled data from the server
+// configuration
+// If we fail here we exit the program with an error
+func (p *Pass) GetConfigForAddon(name string, obj interface{}) error {
+	if _, ok := p.sv.cfg.Addons[name]; !ok {
+		log.Fatalf("Unable to find addon config for %s: %v", name, obj)
+	}
+
+	if err := json.Unmarshal(p.sv.cfg.Addons[name], &obj); err != nil {
+		log.Fatalf("Unable to unmarshal addon config: %v", err)
+	}
+
+	return nil
+}
+
+// resolveClientIP tries to figure out the actual user IP address
+//
+// For trusted proxies it uses the configured TrustedProxies array to check if
+// the RemoteAddr is a proxy we trust and then check the configured
+// TrustedProxyIPHeader for the actual user IP address
+//
+// Othervise, we consider the RemoteAddr to be the user
+func resolveClientIP(rv *requestVariables) {
+	var ipString string
+	var err error
+	if ipString, _, err = net.SplitHostPort(rv.r.RemoteAddr); err != nil {
+		log.Printf("Failed net.SplitHostPort failed on %s", rv.r.RemoteAddr)
+	}
+
+	ip := net.ParseIP(ipString)
+
+	if isTrustedProxy := helpers.IPNetArrayContainsIP(rv.sv.trustedProxies, ip); isTrustedProxy {
```

Comment:
Can the list of trusted proxies be stored in a simple `map[net.IP]bool`? Lookups can be simple `_, isTrusted := trustedProxies[ip]`.

Reply:
Does not work since `IPNetArrayContainsIP` supports CIDR notation.